CMakeLists.txt

```diff
@@ -1151,6 +1152,7 @@ add_library(libtsutil SHARED
         lib/ts/Tokenizer.h
         lib/ts/Trie.h
         lib/ts/TsBuffer.h
+	lib/ts/ts_meta.h
```

Comment:
What is this for?

Reply:
There just a few comments in that header file to explain what it does...

It was part of some work that ended up in the bit bucket. This enabled optional types / functions in the template parameter `H`. The point was to allow overrides of certain deduced types. For example, `H::value_type` would override the type deduced from the link accessor methods but would not be required. Unfortunately the additional implementation effort to make that work was, IMHO, too much work for not enough return - the accessors in `H` can handle that, if needed, more easily. Mostly I forgot this was still here, but I do have some other use cases planned and I might well still use it for this work, so I'd rather leave it in. I suppose I could split that out in to a separate commit.

Reply:
I spent my precious time to find places that require this :-)